src/components/TokenDetailAction.js

```diff
@@ -0,0 +1,438 @@
+              <input required type="number" ref={this.amount} step={hathorLib.helpers.prettyValue(1)} min={hathorLib.helpers.prettyValue(1)} placeholder={hathorLib.helpers.prettyValue(0)} className="form-control" />
+            </div>
+          </div>
+          <div className="form-group d-flex flex-row align-items-center">
+            <div className="form-check">
+              <input className="form-check-input" type="checkbox" ref={this.createAnother} id="keepMint" defaultChecked={true} />
+              <label className="form-check-label" htmlFor="keepMint">
+                Create another melt output for you?
+              </label>
+            </div>
+          </div>
+        </div>
+      )
+    }
+
+    return <TokenAction renderForm={renderForm} title='Melt tokens' buttonName='Go' validateForm={this.melt} onPinSuccess={this.executeMelt} {...this.props} />
+  }
+}
+
+
+class TokenDelegate extends React.Component {
+  constructor(props) {
+    super(props);
+
+    // Reference to address input
+    this.delegateAddress = React.createRef();
+    // Reference to create another authority output checkbox
+    this.delegateCreateAnother = React.createRef();
+  }
+
+  /**
+   * Execute the delegate of outputs
+   *
+   * @param {string} pin PIN user wrote on modal
+   *
+   * @return {Object} Object with promise and success message
+   */
+  executeDelegate = (pin) => {
+    const output = this.props.authorityOutputs[0];
+    const type = this.props.action === 'delegate-mint' ? 'Mint' : 'Melt';
+    const promise = hathorLib.tokens.delegateAuthority(output.tx_id, output.index, output.decoded.address, this.props.token.uid, this.delegateAddress.current.value, this.delegateCreateAnother.current.checked, type.toLowerCase(), pin);
+    return { promise, message: `${type} output delegated!`};
+  }
+
+  render() {
+    const renderForm = () => {
+      return (
+        <div>
+          <div className="row">
+            <div className="form-group col-6">
+              <label>Address</label>
+              <input required ref={this.delegateAddress} type="text" className="form-control" />
+            </div>
+          </div>
+          <div className="form-group d-flex flex-row align-items-center">
+            <div className="form-check">
+              <input className="form-check-input" type="checkbox" ref={this.delegateCreateAnother} id="keepAuthority" defaultChecked={true} />
+              <label className="form-check-label" htmlFor="keepAuthority">
+                Create another {this.props.action === 'delegate-mint' ? 'mint' : 'melt'} output for you?
+              </label>
+            </div>
+          </div>
+        </div>
+      )
+    }
+
+    const title = `Delegate ${this.props.action === 'destroy-mint' ? 'Mint' : 'Melt'}`;
+
+    return <TokenAction renderForm={renderForm} title={title} buttonName='Delegate' onPinSuccess={this.executeDelegate} {...this.props} />
+  }
+}
+
+
+class TokenDestroy extends React.Component {
```

Comment:
Add description of what the component does. In this case, it is used to destroy mint/melt authorities, right?

Reply:
Done